File: backend/app/Models/SessionParticipant.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class SessionParticipant extends Model
{
    use HasFactory;

    public const ROLE_HOST = 'host';
    public const ROLE_GUEST = 'guest';

    protected $fillable = [
        'session_id',
        'user_id',
        'device_id',
        'role',
        'lock_confirmed_at',
        'end_confirmed_at',
        'last_heartbeat_at',
        'last_integrity_state',
        'protection_active',
    ];

    protected function casts(): array
    {
        return [
            'lock_confirmed_at' => 'datetime',
            'end_confirmed_at' => 'datetime',
            'last_heartbeat_at' => 'datetime',
            'last_integrity_state' => 'array',
            'protection_active' => 'boolean',
        ];
    }

    public function session(): BelongsTo
    {
        return $this->belongsTo(Session::class);
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function device(): BelongsTo
    {
        return $this->belongsTo(Device::class);
    }
}

File: ios/MutualOffline/Models/SessionParticipant.swift
import Foundation

enum ParticipantRole: String, Codable {
    case host
    case guest
}

struct SessionParticipant: Codable, Identifiable, Hashable {
    let id: Int
    let sessionId: Int?
    let userId: Int
    let role: ParticipantRole
    let lockConfirmedAt: Date?
    let endRequestedAt: Date?
    let endConfirmedAt: Date?
    let deviceUuid: String?
    let lastHeartbeatAt: Date?
    let user: User?

    enum CodingKeys: String, CodingKey {
        case id
        case sessionId = "session_id"
        case userId = "user_id"
        case role
        case lockConfirmedAt = "lock_confirmed_at"
        case endRequestedAt = "end_requested_at"
        case endConfirmedAt = "end_confirmed_at"
        case deviceUuid = "device_uuid"
        case lastHeartbeatAt = "last_heartbeat_at"
        case user
    }
}
